sia/src/rhp/rpc.rs

```diff
@@ -0,0 +1,613 @@
+        let mut state = Params::new().hash_length(32).to_state();
+        self.accounts.encode(&mut state).unwrap();
+        self.target.encode(&mut state).unwrap();
+        self.contract_id.encode(&mut state).unwrap();
+        revision_number.encode(&mut state).unwrap();
+        state.finalize().into()
+    }
+}
+
+/// RPCReplenishAccountsResponse contains the host's response to the replenish
+/// request.
+///
+/// The renter should verify the deposits and construct a revision
+/// transferring the funds.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCReplenishAccountsResponse {
+    pub deposits: Vec<AccountDeposit>,
+}
+
+/// RPCReplenishAccountsSecondResponse contains the renter's signature for the
+/// contract revision replenishing the accounts.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCReplenishAccountsSecondResponse {
+    pub renter_signature: Signature,
+}
+
+/// RPCReplenishAccountsThirdResponse contains the host's signature for the
+/// contract revision replenishing the accounts.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCReplenishAccountsThirdResponse {
+    pub host_signature: Signature,
+}
+
+/// RPCVerifySectorRequest is the request type for verifying the host
+/// is storing a sector.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCVerifySectorRequest {
+    pub prices: HostPrices,
+    pub token: AccountToken,
+    pub root: Hash256,
+    pub leaf_index: u64,
+}
+
+/// RPCVerifySectorResponse contains a proof that the host is storing a
+/// sector.
+#[derive(Debug, PartialEq, SiaEncode, SiaDecode, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCVerifySectorResponse {
+    pub proof: Vec<Hash256>,
+    pub leaf: Leaf,
+}
+
+/// RPCFundAccountsRequest is the request type for funding accounts
+/// with Siacoin deposits.
+///
+/// RPCReplenishAccounts should be preferred
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCFundAccountsRequest {
+    pub contract_id: FileContractID,
+    pub deposits: Vec<AccountDeposit>,
+    pub renter_signature: Signature,
+}
+
+/// RPCFundAccountsResponse contains the host's signature and new
+/// balance after funding the accounts.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCFundAccountsResponse {
+    pub balances: Vec<Currency>,
+    pub host_signature: Signature,
+}
+
+/// RPCError is the error type returned by the RPC server.
+#[derive(Debug, PartialEq, Serialize, Deserialize, SiaEncode, SiaDecode)]
+#[serde(rename_all = "camelCase")]
+pub struct RPCError {
+    pub code: u8,
```

Comment:
(nit) enum for error codes? Depending on how annoying that is to encode

Reply:
will follow up since this is getting huge